workshop/hello-llama/porter.yaml

```diff
@@ -17,18 +17,18 @@ install:
       command: ./helpers.sh
       arguments:
         - install
-        - "{{ bundle.parameters.name }}"
+        - "'{{ bundle.parameters.name }}'"
```

Comment:
@carolynvs 
We investigated this further and concluded it has to do with how the string is escaped in YAML. This probably requires an update to documentation but additionally requires code changes to existing bundles.

For the scope of this issue, would a change of documentation and a code change to the llama bundle be sufficient or do you see an alternative route?
Our proposal is to extend this [documentation](https://porter.sh/quickstart/parameters/#create-a-parameter-set) to include a note regarding how to handle parameters that include a space. We are unsure about the functionality behind the `porter parameters generate` command and whether an update here is required.
Lastly, what we see as being out of scope is updating all the other places where YAML variables could include space and therefore could face a similar problem.

Would you agree?

Reply:
Hmm, I wasn't expecting the wrapping quotes to matter here, since we are parsing this and converting it into a an array of arguments in Go, e.g. ["./helpers.sh", "install", "my parameter"]. I would have expected calling a bash script in Go and passing an argument with a space to work, without the quotes. If you can look into that more, that would be great, otherwise I'll try to when I have time.

Reply:
From our investigation, we see that the strings get passed separately in the bash script, for example when we passed $1, $2, and so on in the `/helpers.sh` that was as many words as we could get. We also tried $@ which we did not think was very optimal because this would catch all arguments even when not unique to the command. It looks like the values get read in the YAML file in the same manner.
I have had a couple of other engagements of recent and will investigate this further. In the meantime, if you get a chance to look at this and give me more pointers, that will be great. We came across https://github.com/getporter/porter/blob/release/v1/pkg/exec/builder/execute.go#L102, especially the `splitCommand(args)` line, we will look at this next.

Reply:
We looked further into the `[splitCommand(args)](https://github.com/getporter/porter/blob/release/v1/pkg/exec/builder/execute.go#L128)` function which was added in this [commit](https://github.com/getporter/porter/pull/931). This does look like a possible culprit but not something we will like to change. We are potentially missing some context, is there a chance that you @carolynvs could give us some context? Thanks in advance!

Reply:
Adding a note here, that I spotted that the escaping in the `porter.yaml` was being done in these [example bundles](https://github.com/getporter/example-bundles/blob/main/azure-keyvault/porter.yaml#L20) for flags, still not sure if they are handled differently somehow

Reply:
I can confirm that this `splitCommand(args)` function is responsible for dividing the argument (I commented it out and rebuilt and now see the full name as expected) Perhaps there is somewhere earlier in the code where we can enforce this string is not escaped and therefore split when this function runs.

Reply:
Thanks for narrowing down where the bug is!

Reply:
Awesome congrats @Simpcyclassy :tada: on your awesome work here and be along the way in solving your first open-source bug :bug: :raised_hands: In our next session we can look at what propositions we can make for a solution.

Reply:
I just ran into this problem with a different flag in Porter. It sounds like we are using cobra incorrectly and that's why it's splitting on the comma

https://github.com/spf13/cobra/issues/661#issuecomment-377684634

Reply:
Let me know if you need some help figuring out what needs to be changed in the cmd/porter package files. I believe switching any use of StringSliceVar to StringArrayVar fix it.

Reply:
@carolynvs we went with the approach of switching our StringSliceVar to StringArrayVar but it appears to not have made the change we imagined. We will keep working on this but we need to take a break this next week as @Simpcyclassy is traveling. We will for sure pick it up when she is back. We feel so close!

Reply:
@sleepypioneer @Simpcyclassy I just realized that I completely messed up and linked to the StringArray cobra fix on the wrong pull request! I'm so sorry! 

Both (parsing the comma and handling whitespace) were things that needed to be fixed, but I messed up and commented on the wrong pull request. I really didn't mean to derail you.

It sounds like you had a fix working for the space problem too? I think you were on the right track with `splitCommand`. You really are close, I just accidentally threw at you a second (unreleated) bug on this pull request.

You can either fix both on this PR, or just skip the one failing test (tests/integration/install_test.go) and we can merge what you have here (which does fix the comma bug), then fix the parameter space bug in a second pull request. Whatever works easier for you.

Reply:
Oh good to know, I didn't catch that there was a second bug for commas. We can look again at the `splitCommand` function, we stopped looking there, after seeing about the cobra method misuse :see_no_evil: Thanks for your patience, we are super excited to be contributing. We only meet once a week so it takes us a little time but as you said we are really close! Thanks for writing and letting us know!!

Reply:
@carolynvs this is the issue for the comma https://github.com/getporter/porter/issues/1000 I will write there that this PR will add the fix.

I am not too sure how we can best change the behavior, from reading the code for `splitCommand`, it is working as intended. We either need to wrap our param in extra quotes or we have to skip this processing for parameters but I am concerned that is undoing the reason this was implemented in the first place?

Reply:
Adding a little extra context.

If we leave the `splitCommand(args)` in this is the output we see for first a param with quotes around it and secondly a param without quotes:

```
greenmachine@pop-os:/tmp/hello-world$ porter install --param name="demo time" --force
executing install action from porter-hello (installation: /porter-hello)
Install Hello World
Hello demo
execution completed successfully!

greenmachine@pop-os:/tmp/hello-world$ porter install --param name=demo time --force
executing install action from porter-hello (installation: /time)
Install Hello World
Hello demo
execution completed successfully!
```

Now the same with the call to `splitCommand` removed:

```
greenmachine@pop-os:/tmp/hello-world$ porter install --param name="demo time" --force
executing install action from porter-hello (installation: /porter-hello)
Install Hello World
Hello demo time
execution completed successfully!

greenmachine@pop-os:/tmp/hello-world$ porter install --param name=demo time --force
executing install action from porter-hello (installation: /time)
Install Hello World
Hello demo
execution completed successfully!
```
We see that removing the `splitCommand` call makes things behave as expected, but I am still unsure if there is an edge case we miss that makes it unwise to remove this. We could move the call up and run it only for flags (I haven't dug into how flags use this or are expected to behave) It seems at least to me that what this command was trying to solve is no longer an issue for params.

Reply:
I am also thinking that in this case arguments != parameters, so we may want to only have an effect on parameters that are passed through as other arguments may still require this behavior from `splitCommand` looking to see where we could do that.